api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -291,6 +295,64 @@ scalar UserAssignedIdentityResourceId
     }
   ]>;
 
+/** A representation of a KeyVault Secret. */
+model KmsKey {
+  /** vaultName is the name of the keyvault that contains the secret */
+  @maxLength(255)
+  @minLength(1)
+  vaultName: string;
+
+  /** name is the name of the keyvault key used for encrypt/decrypt */
+  @maxLength(255)
+  @minLength(1)
+  name: string;
+
+  /** version contains the version of the key to use */
+  @maxLength(255)
+  @minLength(1)
+  version: string;
+}
+
+/** A KMS Key that is immutable. */
+@withVisibilityFilter(#{ all: #[Lifecycle.Create, Lifecycle.Read] })
+model NonUpdatableKmsKey {
+  ...KmsKey;
+}
+
+/** The ETCD settings and configuration options. */
+model EtcdProfile {
+  /** ETCD Data Encryption settings. */
+  dataEncryption: EtcdDataEncryptionProfile;
+}
+
+/** The ETCD data encryption settings. */
+model EtcdDataEncryptionProfile {
+  /** Specify whether the key is customer or platform managed. */
+  managementMode: "customerManaged" | "platformManaged" | string;
+
+  /** Specify customer managed encryption key details. */
+  customerManagedConfig?: CustomerManagedConfig;
+}
+
+/** Customer managed encryption key details. */
+model CustomerManagedConfig {
+  /** The encryption type used. */
+  encryptionType: "kms" | string;
```

Comment:
Should this be marked as extensible enum?

Reply:
Switching it only generates a small change in the 'openapi.json' so I was planning to leave it simple until we need it to be more. 

I don't feel strongly about it.

Reply:
I think we need it so it is prepared for extension without being considered a breaking change in the uture